### PR TITLE
Update Teams ScubaGear Rego code to align with revised baselines

### LINKED ISSUE #396: Update Teams ScubaGear code to align to revised baseline
# 💡 Summary #

We made significant structural changes to the Teams baseline during Emerald. The ScubaGear code the policies  needs to be revised to align with the new baseline document. The order of couple policies have changed, while another policy has been divided into sub-policies. This issues to align the structure (policy IDs, group titles, etc.) with the baseline document. 

Any new policies or policies that need a change of logic in the Rego code would be addressed through separate issues. 

## Implementation notes ##
- [ ] Review the new Teams baseline
- [ ]  Update code to place the policies into their appropriate new groups
- [ ]  Reorder and Renumber the policies to match the baseline
- [ ]  Delete any policies in the code that are no longer in the baseline document

## Acceptance criteria ##
ScubaGear output for the Teams product should align with Teams baseline document - in terms of policy number, order, IDs, titles etc.

### LINKED ISSUE #499: Teams policy 2.3: split, rearrange and delete it into new MS.TEAMS.1.3 to 1.5 policies
# 🐛 Summary #

The existing policy 2.3 about Teams meetings has been overhauled with some of redundant checks deleted, others being rearranged and added an additional test. This issues is to address all changes related to old teams policy 2.3 and converting it into 1.3-1.5 policies of new Team policy group 1. 

## To reproduce ##

Reference the MS.TEAMS SCB's for the Emerald Release - refactor the code for policy 2.3 for the new policy settings. 

## Expected behavior ##

The Teams report should show result for policies 1.3-1.5 correctly.

### PR DESCRIPTION
**Pull request by nanda-katikaneni**
## 🗣 Description ##

- Updates Teams Rego code to align to the revised baseline
- Improvements to the accuracy of the Rego check for Policy Groups 1, 2, and 5.
- Does not fully address #452 

## 💭 Motivation and context ##

Closes #396 
Closes #499 

## 🧪 Testing ##

Tested on G5 and E5 tenants.

## ✅ Pre-approval checklist ##

- [ ] This PR has an informative and human-readable title.
- [ ] Changes are limited to a single goal - *eschew scope creep!*
- [ ] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [ ] All relevant type-of-change labels have been added.
- [ ] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [ ] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [ ] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: Testing/Unit/Rego/Teams/TeamsConfig_02-test.rego [hunk 1]
```diff
@@ -2,11 +2,219 @@ package teams
 import future.keywords
 
 
-#
-# Policy 1
+#--
+# Policy MS.TEAMS.2.1v1
```
Comment (crutchfield) [inline Testing/Unit/Rego/Teams/TeamsConfig_02-test.rego:6]: Revisit test after addressing general policy question on MS.TEAMS.2.1v1 implementation.

File: Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego [hunk 2]
```diff
@@ -273,17 +276,17 @@ test_PrivateCatalogAppsType_Incorrect_V1 if {
 
     count(RuleOutput) == 1
     not RuleOutput[0].RequirementMet
-    RuleOutput[0].ReportDetails == "1 meeting policy(ies) found that allow custom apps by default: Global"
+    RuleOutput[0].ReportDetails == "1 meeting policy(ies) found that does not restrict installation of third-party apps by default: Global"
```
Comment (crutchfield) [inline Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego:279]:
```suggestion
    RuleOutput[0].ReportDetails == "1 meeting policy(ies) found that does not restrict installation of custom apps by default: Global"
```

File: Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego [hunk 3]
```diff
@@ -292,49 +295,35 @@ test_PrivateCatalogAppsType_Incorrect_V2 if {
 
     count(RuleOutput) == 1
     not RuleOutput[0].RequirementMet
-    RuleOutput[0].ReportDetails == "1 meeting policy(ies) found that allow custom apps by default: Tag:TestPolicy"
+    RuleOutput[0].ReportDetails == "1 meeting policy(ies) found that does not restrict installation of third-party apps by default: Tag:TestPolicy"
```
Comment (crutchfield) [inline Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego:298]:
```suggestion
    RuleOutput[0].ReportDetails == "1 meeting policy(ies) found that does not restrict installation of custom apps by default: Tag:TestPolicy"
```

File: Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego [hunk 4]
```diff
@@ -292,49 +295,35 @@ test_PrivateCatalogAppsType_Incorrect_V2 if {
...
         ]
     }
-    
+
     RuleOutput := [Result | Result = Output[_]; Result.PolicyId == PolicyId]
 
     count(RuleOutput) == 1
     not RuleOutput[0].RequirementMet
-    startswith(RuleOutput[0].ReportDetails, "2 meeting policy(ies) found that allow custom apps by default: ")
+    startswith(RuleOutput[0].ReportDetails, "2 meeting policy(ies) found that does not restrict installation of third-party apps by default: ")
```
Comment (crutchfield) [inline Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego:325]:
```suggestion
    startswith(RuleOutput[0].ReportDetails, "2 meeting policy(ies) found that does not restrict installation of custom apps by default: ")
```

**Review by crutchfield — CHANGES_REQUESTED** [review]
Looks good overall.  I do have a question on MS.TEAMS.2.1.v1 as to teh intent of the policy.  Other than that just some minor comments/change requests.

File: Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego (hunk 2 above)
Comment (Dylan-MITRE) [reply Testing/Unit/Rego/Teams/TeamsConfig_05-test.rego]: good catch, fixed along with the other two regarding policy 5 unit test

File: Rego/TeamsConfig.rego [hunk 5]
```diff
@@ -151,8 +195,67 @@ tests[{
...
+	"RequirementMet" : false
+}] {
+	count(input.broadcast_policies) == 0
+}
+#--
+
+#--
+# MS.TEAMS.2.1v1
 #--
 ExternalAccessConfig[Policy.Identity] {
```
[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply Rego/TeamsConfig.rego:259]: The policy is to restrict the external domains the users have access to - from the default which allows all 'external' domains to allow only specific domains (for the allowed domains property). I tested both positive and negative cases and results are correct. Am I missing something?
Test Case: to pass with "only specific domains allowed" :
Screenshot 2023-09-28 at 11 34 47 AM

Test Case: to fail with "all external domains allowed":
Screenshot 2023-09-28 at 11 34 58 AM

File: Testing/Unit/Rego/Teams/TeamsConfig_01-test.rego [hunk 6]
```diff
@@ -0,0 +1,555 @@
...
+# Policy MS.TEAMS.1.5v1
+#--
+test_meeting_policies_Correct if {
+    PolicyId := "MS.TEAMS.1.5v1"
+
+    Output := tests with input as {
+        "meeting_policies": [
+            {
+                "Identity": "Tag:CustomPolicy",
+                "AllowPSTNUsersToBypassLobby": false
```
[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply Testing/Unit/Rego/Teams/TeamsConfig_01-test.rego]: accepted the suggestion and committed.

File: Testing/Unit/Rego/Teams/TeamsConfig_01-test.rego [hunk 7]
```diff
@@ -0,0 +1,555 @@
...
+}
+
+test_AllowPSTNUsersToBypassLobby_Incorrect if {
+    PolicyId := "MS.TEAMS.1.5v1"
+
+    Output := tests with input as {
+        "meeting_policies": [
+            {
+                "Identity": "Tag:CustomPolicy",
+                "AllowPSTNUsersToBypassLobby": true
```
[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply Testing/Unit/Rego/Teams/TeamsConfig_01-test.rego]: accepted the suggestion.

File: Rego/TeamsConfig.rego [hunk 8]
```diff
@@ -522,12 +504,12 @@ tests[{
 }
 #--
 
-#
-# MS.TEAMS.11.2v1
+#--
+# MS.TEAMS.6.2v1
```
[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply Rego/TeamsConfig.rego:507]: The reference is to ScubaGear itself (running the Defender product) - made changes to the comment that appear on the report for policies 6-8, to make it more explicit about running ScubaGear for Defender. Good catch.

File: Rego/TeamsConfig.rego [hunk 9]
```diff
@@ -176,30 +279,7 @@ tests[{
...
-}] {
-	Policies := MeetingsNotAllowingAnonJoin
-	String := "meeting policy(ies) found that don't allow anonymous users to join meetings:"
-	Status := count(Policies) == 0
-}
-#--
-
-#
-# MS.TEAMS.5.1v1
+# MS.TEAMS.2.2v1
```
[the remark this answers is not in the document]
Comment (nanda-katikaneni) [reply Rego/TeamsConfig.rego:281]: Indentation fixed.

**Review by crutchfield — APPROVED** [review]
Approved

**Review by ahuynhMITRE — APPROVED** [review]

**Review by rgbrow1949 — APPROVED** [review]
Approved.

**Pull request merged**